test/deployframework/local.go

```diff
@@ -0,0 +1,212 @@
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	ansibleRunnerPath                   = "/tmp/ansible-operator/runner"
+	meteringOperatorName                = "metering-ansible-operator"
+	meteringOperatorLogName             = "metering-operator.log"
+	reportingOperatorLogName            = "reporting-operator.log"
+	destKubeConfigPath                  = "/kubeconfig"
+	reportingOperatorTLSDirName         = "reporting-operator-tls"
+	runReportingOperatorLocalScriptName = "run-reporting-operator-local.sh"
+	cleanupScriptName                   = "run-test-cleanup.sh"
+
+	enableDebug        = true
+	disableOCPFeatures = false
+)
+
+// LocalCtx holds all the necessary information to run e2e tests locally
+type LocalCtx struct {
+	Namespace                      string
+	BasePath                       string
+	KubeConfigPath                 string
+	HackScriptPath                 string
+	MeteringOperatorImage          string
+	ReportingAPIURL                string
+	MeteringOperatorCmd            *exec.Cmd
+	ReportingOperatorCmd           *exec.Cmd
+	ReportingOperatorTLSPath       string
+	RunReportingOperatorScriptPath string
+	Logger                         logrus.FieldLogger
+}
+
+// RunMeteringOperatorLocal is a method that runs the metering-operator locally
+func (lc *LocalCtx) RunMeteringOperatorLocal() error {
+	lc.MeteringOperatorCmd = exec.Command(
+		"docker", "run",
+		"--rm",
+		"-u", "0:0",
+		"--name", meteringOperatorContainerName,
+		"-v", ansibleRunnerPath,
+		"-v", fmt.Sprintf("%s:%s", lc.KubeConfigPath, destKubeConfigPath),
+		"-e", "KUBECONFIG="+destKubeConfigPath,
+		"-e", "OPERATOR_NAME="+meteringOperatorName,
+		"-e", "POD_NAME="+meteringOperatorName,
+		"-e", "WATCH_NAMESPACE="+lc.Namespace,
+		"-e", "ENABLE_DEBUG="+strconv.FormatBool(enableDebug),
+		"-e", "DISABLE_OCP_FEATURES="+strconv.FormatBool(disableOCPFeatures),
+		lc.MeteringOperatorImage,
+	)
+
+	lc.Logger.WithFields(logrus.Fields{
+		"command": lc.MeteringOperatorCmd.Args,
+	}).Infof("Running the metering-operator docker container")
+
+	logFile, err := ioutil.TempFile(lc.BasePath, meteringOperatorLogName)
+	if err != nil {
+		return fmt.Errorf("failed to create the metering-operator container output log file: %v", err)
+	}
+
+	lc.Logger.Infof("Storing the metering-operator container logs to the '%s' path", logFile.Name())
+
+	lc.MeteringOperatorCmd.Stdout = logFile
+	lc.MeteringOperatorCmd.Stderr = logFile
+
+	err = lc.MeteringOperatorCmd.Start()
+	if err != nil {
+		return fmt.Errorf("failed to run the metering-operator container: %v", err)
+	}
+
+	return nil
+}
+
+// RunReportingOperatorLocal is a method that runs the reporting-operator locally
+func (lc *LocalCtx) RunReportingOperatorLocal(apiListenPort, metricsListenPort, pprofListenPort int, token string) error {
+	var err error
+
+	// TODO: should this still be a tmpdir in the case where tests are interrupted early?
+	lc.ReportingOperatorTLSPath, err = ioutil.TempDir(lc.BasePath, reportingOperatorTLSDirName)
```

Comment:
We don't want to hardcode directories, ever. If this is an issue it should be adapted to support using an existing directory instead of creating a tempdir. 

Additionally, it may be best if we simply replace the logic in `run-reporting-operator-local.sh` with Go code since there's so many background tasks, it would be way simpler to do it in Go.